pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -535,3 +536,13 @@ def teardown(self) -> None:
             self.lightning_module.cpu()
             # clean up memory
             torch.cuda.empty_cache()
+
+    def load_checkpoint(self, checkpoint_path: _PATH) -> Dict[str, Any]:
+        rank_zero_info(f"DistributedDataParallel has {self.num_processes} processes. Serializing to avoid CPU OOMs.")
+        for current_worker in range(self.num_processes):
+            if self.local_rank == current_worker:
+                checkpoint = super().load_checkpoint(checkpoint_path)
+                del checkpoint["state_dict"]
```

Comment:
load_model_state_dict should check if `"state_dict"` as a key is in the checkpoint dict. otherwise it'll fail on lookup

Reply:
what's the expected behavior if `"state_dict"` key does not exist?

Reply:
got it!

Reply:
- the plugin can track whether it was the entity that deleted the key from the checkpoint in `load_checkpoint` . if it did, and there's no `"state_dict"` found when calling `load_model_state_dict` the plugin can simply return early
- if it didn't, then the plugin could raise an exception about the key not found as this is unexpected

Reply:
hm do you mean something like the below edit?

Reply:
yup, but this doesn't need to be on the base training type plugin. it can be initialized on the plugins which actually serialize checkpoints